plotly/offline.py

```diff
@@ -0,0 +1,149 @@
+""" Plotly Offline
+    A module to use Plotly's graphing library with Python
+    without connecting to a public or private plotly enterprise
+    server.
+"""
+from __future__ import absolute_import
+
+from plotly import utils
+import uuid
+import json
+import os
+import requests
+
+
+def download_plotlyjs(download_url, plotlyjs_dir='~/.plotly/plotlyjs'):
+    plotlyjs_path = os.path.expanduser(
+        os.path.join(*'~/.plotly/plotlyjs'.split('/')))
+
+    if not os.path.exists(plotlyjs_path):
+        os.makedirs(plotlyjs_path)
+
+    res = requests.get(download_url + '/sourcefiles.json')
+    res.raise_for_status()
+
+    with open(os.path.join(plotlyjs_path, 'sourcefiles.json'), 'w') as f:
```

Comment:
they contain the list of plotly.js dependencies:
```
{
    "files": ["dependencies/d3.v3.min.js", "dependencies/jquery-latest.js", "dependencies/typedarray.js", "plotly.min.js"]
}
```

that way, if we change the dependencies in the future we don't have to change anything on the python lib side. they just explain which files are necessary to include and to download